doc/specs/#885 - winrt Terminal Settings.md

```diff
@@ -0,0 +1,251 @@
+
+To get around this issue, when `CascadiaSettings` encounters a serialization error, it must internally record
+ any pertinent information for that error, and return the simple `CascadiaSettings` as if nothing happened.
+ The consumer must then call `CascadiaSettings::GetErrors()` and `CascadiaSettings::GetWarnings()` to properly
+ understand whether an error ocurred and how to present that to the user.
+
+
+#### TerminalApp: Loading and Reloading Changes
+
+TerminalApp will construct and reference a `CascadiaSettings settings` as follows:
+- TerminalApp will have a global reference to the "settings.json" filepath
+- construct an `CascadiaSettings` using `CascadiaSettings("settings.json")`. This builds an `CascadiaSettings`
+   from the "defaults.json" file data (which is already compiled as a string literal) 
+   and layers the settings.json data on top of it.
+- check for errors/warnings, and handle them appropriately
+
+This will be different from the current model which has the settings.json path hardcoded, and is simplified
+ to a `LoadAll()` call wrapped in error handlers.
+
+**NOTE:** This model allows us to layer even more settings files on top of the existing Terminal Settings
+ Model, if so desired. This could be helpful when importing additional settings files from an external location
+ such as a marketplace.
+
+When TerminalApp detects a change to settings.json, it'll repeat the steps above. We could cache the result from
+ constructing an `CascadiaSettings` from "defaults.json" data to improve performance.
+
+
+#### TerminalControl: Acquiring and Applying the Settings
+
+At the time of writing this spec, TerminalApp constructs `TerminalControl.TerminalSettings` WinRT objects
+ to expose `IControlSettings` and `ICoreSettings` to any hosted terminals. In moving `IControlSettings`
+ and `ICoreSettings` down to the TerminalControl layer, TerminalApp can now have better control over
+ how to expose relevant settings to a TerminalControl instance.
+
+`TerminalSettings` (which implements `IControlSettings` and `ICoreSettings`) will be moved to
+ TerminalApp and act as a bridge connecting `CascadiaSettings` to the TermControl. It will operate
+ very similarly as it does today. On construction of the TermControl or hot-reload,
+ `TerminalSettings` will be constructed by copying the relevant values of `CascadiaSettings`.
+ Then, it will be passed to TermControl (and TermCore by extension).
+
+
+## UI/UX Design
+
+N/A
+
+## Capabilities
+
+### Accessibility
+
+N/A
+
+### Security
+
+N/A
+
+### Reliability
+
+N/A
+
+### Compatibility
+
+N/A
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+### Deserialization
+
+After deserializing the settings, injecting the new json into settings.json
+ should not remove the existing comments or formatting.
+
+The deserialization process takes place right after comparing the `settingsSource` and `settingsClone` objects.

```

Comment:
I'm not sure the `settingsSource` and `settingsClone` objects are still referenced in this doc...

Reply:
Okay so they're below this point. Maybe this whole section should be moved to a sub-header of that one?